contracts/StorageManager.sol

```diff
@@ -46,48 +49,63 @@ contract StorageManager {
     // offerRegistry stores the open or closed Offer for provider.
     mapping(address => Offer) public offerRegistry;
 
+    // maps the tokenAddresses which can be used with this contract. By convention, address(0) is the native token.
+    mapping(address => bool) public isWhitelistedToken;
+
     event TotalCapacitySet(address indexed provider, uint128 capacity);
-    event BillingPlanSet(address indexed provider, uint64 period, uint64 price);
+    event BillingPlanSet(address indexed provider, address token, uint64 period, uint64 price);
     event MessageEmitted(address indexed provider, bytes32[] message);
 
     event NewAgreement(
-        bytes32 agreementReference,
-        bytes32[] dataReference,
+        bytes32[] dataReference,	
```

Comment:
can you explain this field? is this because you reached maximum size allow of info to emit in an Event? so to Obtain the `token` used for an agreement we would need to make a call after getting the event, right?

Reply:
See: https://github.com/rsksmart/rif-marketplace-storage/pull/113#discussion_r471940742
Probably made a mistake here. Pls have a look after I commit changes if you agree